packages/runtime-core/src/component.ts

```diff
@@ -988,9 +988,12 @@ export function formatComponentName(
 ): string {
   let name = getComponentName(Component)
   if (!name && Component.__file) {
-    const match = Component.__file.match(/([^/\\]+)\.\w+$/)
-    if (match) {
-      name = match[1]
+    const match = Component.__file.match(/(\/(\w+))?(\/(\w+))\.\w+$/)
```

Comment:
The regex doesn't consider the separator of Windows `\`, like `D:\Test\index.vue`

Reply:
https://github.com/vuejs/core/pull/7154#issue-1451785716